calico-enterprise/operations/monitor/metrics/recommended-metrics.mdx

```diff
@@ -0,0 +1,514 @@
+| Priority level                   | Recommended.                                                 |
+
+## Felix time-based metrics
+
+### Data plane apply time quantile 0.5/0.9/0.99
+
+| Data plane apply time quantile 0.5/0.9/0.99 |                                                              |
+| ------------------------------------------ | ------------------------------------------------------------ |
+| Metric                                     | *felix_int_dataplane_apply_time_seconds{quantile="0.5"}/{quantile="0.9"}/{quantile="0.99"}* |
+| Explanation                                | Time in seconds that it takes to apply a data plane update viewed at the median, 90th percentile, and 99th percentile. |
+| Threshold value recommendation             | Thresholds will vary depending on cluster size and rate of churn.  It is recommended that a baseline be set to determine a normal threshold value. Examples of what has been seen in the field: 3-node test cluster averaging 100ms 1000-node 15x federated cluster averaging 30s (potentially larger at the start of business day). |
+| Threshold breach symptoms                  | Large time-to-apply values will cause a delay between {{prodname}} network policy commits and enforcement in the dataplane.  This is dependent on how {{prodname}} waiting for kube-proxy to release the iptables lock, which is influenced by the number of services in use. |
+| Threshold breach recommendations           | Increase cluster resources and reduce the number of Kubernetes services if possible. |
+| Priority level                             | Recommended.                                                 |
+
+### Felix route table list seconds quantile 0.5/0.9/0.99
+
+| Felix route table list seconds quantile 0.5/0.9/0.99 |                                                              |
+| ---------------------------------------------------- | ------------------------------------------------------------ |
+| Metric                                               | *felix_route_table_list_seconds{quantile="0.5"}/{quantile="0.9"}/{quantile="0.99"}* |
+| Explanation                                          | Time to list all the interfaces during a resync viewed at the median, 90th percentile, and 99th percentile. |
+| Threshold value recommendation                       | Thresholds will vary depending on the number of Cali interfaces per node.  We recommend setting a baseline to determine a normal threshold value. Examples of what has been seen in the field: <br />- <100 cali interfaces averages sub-second<br />- >250 cali interfaces (we expect values in seconds) |
```

Comment:
Think this should be better in recent releases, I fixed the slowness issue that affected larger nodes.

Reply:
Not sure what to change unless we can update the field results.